march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator_controller.py

```diff
@@ -0,0 +1,310 @@
+import os
+
+from numpy_ringbuffer import RingBuffer
+import rospy
+from trajectory_msgs.msg import JointTrajectory
+from urdf_parser_py import urdf
+
+from .model.modifiable_setpoint import ModifiableSetpoint
+from .model.modifiable_subgait import ModifiableSubgait
+
+
+class GaitGeneratorController(object):
+
+    def __init__(self, view):
+        self.view = view
+
+        # Default values
+        self.gait_publisher = None
+        self.set_topic_name(self.view.topic_name_line_edit.text())
+        self.gait_directory = None
+
+        self.playback_speed = 100
+        self.time_slider_thread = None
+        self.current_time = 0
+
+        self.robot = urdf.Robot.from_parameter_server()
```

Comment:
I would make the urdf a parameter for `__init__`, so that you do not need to set up a ROS parameter server and can just inject an arbitrary robot. Then I would move the getting from the parameter server to the `Plugin`

Reply:
You would still need a robot though - and the march4 robot would be